force-app/main/default/classes/IBMWatsonJSONUtil.cls

```diff
@@ -110,16 +113,22 @@ public class IBMWatsonJSONUtil {
               braceCount--;
               Integer lastIndex = replacementList.size() - 1;
               String lastVal = replacementList.get(lastIndex);
-
-              lastVal = lastVal.substring(0, lastVal.length() - 1);
-              replacementList.set(lastIndex, lastVal);
+              
+              // only want to remove trailing commas
+              if (!lastVal.equals('{')) {
```

Comment:
was this a problem introduced by serializing/deserializing `additional_properties`?

Reply:
No I think this was just a bug I hadn't encountered when writing that method originally. It failed dealing with empty objects or arrays.